src/Machine.js

```diff
@@ -238,17 +247,28 @@ class Machine {
   _includeSource(source, context, buffer, once, evaluated) {
 
     // path is an expression, evaluate it
-    const includePath = evaluated ? source : this.expression.evaluate(
+    let includePath = evaluated ? source : this.expression.evaluate(
         source, this._mergeContexts(this._globalContext, context)
-      );
+      ).trim();
 
     // if once flag is set, then check if source has alredy been included
     if (once && this._includedSources.has(includePath)) {
       this.logger.debug(`Skipping source "${includePath}": has already been included`);
       return;
     }
 
-    const reader = this._getReader(includePath);
+    let reader = this._getReader(includePath);
```

Comment:
I'd move all of this logic to FileCahce. The result code should look like:
     this.logger.info(`Including source "${includePath}"`);
      let content = FileCache.read(includePath);